python/lsst/drp/tasks/fit_turbulence.py

```diff
@@ -0,0 +1,586 @@
+        turbulence.
+
+        Parameters
+        ----------
+        inputWcs : `lsst.afw.table.ExposureCatalog`
+            Catalog with WCSs for each detector of the input exposure.
+        inputPositions : `astropy.table.Table`
+            Catalog of input positions with residuals to the current best fit.
+
+        Returns
+        -------
+        outputWcs : `lsst.afw.table.ExposureCatalog`
+            Catalog with WCS after inserting the correction for atmospheric
+            turbulence.
+        """
+
+        visit = inputWcs[0]["visit"]
+
+        inputPositions = inputPositions.get(
+            parameters={
+                "columns": [
+                    "xworld",
+                    "yworld",
+                    "xresw",
+                    "yresw",
+                    "sigpix",
+                    "exposureName",
+                    "xpix",
+                    "ypix",
+                    "deviceName",
+                ]
+            }
+        )
+
+        visitPositions = inputPositions[inputPositions["exposureName"] == str(visit)]
+
+        gpx, gpy, *_ = self.runGP(inputWcs, visitPositions)
+
+        # self.evaluate(gpx, gpy, trainInd, testInd, x, y, dx, dy, residError)
+
+        wcsWithSpline = self.addGPToWcs(gpx, gpy, inputWcs)
+        import ipdb
+
+        ipdb.set_trace()
+        return pipeBase.Struct(outputWcs=wcsWithSpline)
+
+    def runGP(self, inputWcs, positions):
+        """Run Gaussian Processes in tangent plane coordinates.
+
+        Parameters
+        ----------
+        inputWcs : `lsst.afw.table.ExposureCatalog`
+            Catalog with WCSs for each detector of the input exposure.
+        inputPositions : `astropy.table.Table`
+            Catalog of input positions with residuals to the current best fit.
+
+        Returns
+        -------
+        gpx : `treegp.gp_interp.GPInterpolation`
+            Gaussian Processes interpolator for x-direction residuals.
+        gpy : `treegp.gp_interp.GPInterpolation`
+            Gaussian Processes interpolator for y-direction residuals.
+        trainInds : `numpy.ndarray`
+            Array of indices for points used in training.
+        testInds : `numpy.ndarray`
+            Array of indices for points not used in training.
+        """
+        dx = positions["xresw"]
+        dy = positions["yresw"]
+        residError = (positions["sigpix"] * self.config.pixelSize * u.arcsec).to(u.mas).value
+
+        # Get tangent plane coordinates for input points
+        allTPCoords = np.zeros((len(positions), 2))
+        for detector in inputWcs:
+            detId = detector["id"]
+            detWCS = detector.wcs
+            detInd = positions["deviceName"].astype(int) == detId
+            detectorSources = positions[detInd]
+
+            tangentPlaneToSky = detWCS.getFrameDict().getMapping("PIXELS", "IWC")
+            tangentPlaneCoords = tangentPlaneToSky.applyForward(
```

Comment:
If the unit of this in degree here? It needs to be in the same units as min and max sep.